packages/data/src/navigation/reducer.js

```diff
@@ -29,6 +32,100 @@ const reducer = (
 				menuItems: [ ...state.menuItems, ...menuItems ],
 			};
 			break;
+		case TYPES.GET_FAVORITES_FAILURE:
+			state = {
+				...state,
+				requesting: {
+					...state.requesting,
+					getFavorites: false,
+				},
+			};
+			break;
+		case TYPES.GET_FAVORITES_REQUEST:
+			state = {
+				...state,
+				requesting: {
+					...state.requesting,
+					getFavorites: true,
+				},
+			};
+			break;
+		case TYPES.GET_FAVORITES_SUCCESS:
+			state = {
+				...state,
+				favorites,
+				requesting: {
+					...state.requesting,
+					getFavorites: false,
+				},
+			};
+			break;
+		case TYPES.ADD_FAVORITE_FAILURE:
+			state = {
+				...state,
+				error,
+				requesting: {
+					...state.requesting,
+					addTodo: false,
+				},
+			};
+			break;
+		case TYPES.ADD_FAVORITE_REQUEST:
+			state = {
+				...state,
+				requesting: {
+					...state.requesting,
+					addTodo: true,
+				},
+			};
+			break;
+		case TYPES.ADD_FAVORITE_SUCCESS:
+			const newFavorites = ! state.favorites.includes( favorite )
```

Comment:
This check is also performed in the API logic, so perhaps unnecessary?

Reply:
Also, same as my other comment; this request currently returns all favorites (including added) on success, so we could just directly plug in the response.

Reply:
Using all favorites is a possibility, though I'm hesitant to remove the logic around whether or not it's included client-side if we add a single item.

There are still potential for side effects in the state that would make it wise to safe guard this.